src/lib.rs

```diff
@@ -36,25 +37,26 @@ use crate::{
 #[grammar = "minimal.pest"]
 pub struct IdentParser;
 
-pub fn _compile(file: &Path) -> Arc<Node<Named<Commit<Elements>>>> {
-    let file = std::fs::read_to_string(file).unwrap();
-    let mut pairs = IdentParser::parse(Rule::program, &file).unwrap_or_else(|e| panic!("{}", e));
-
-    let prog = Program::parse(pairs.next().unwrap());
+pub fn compile_named(program: Arc<str>) -> Result<Arc<Node<Named<Commit<Elements>>>>, String> {
+    let simfony_program = IdentParser::parse(Rule::program, &program)
+        .map_err(RichError::from)
+        .and_then(|mut pairs| Program::parse(pairs.next().unwrap()))
+        .with_file(program.clone())?;
 
     let mut scope = GlobalScope::new();
-    let simplicity_prog = prog.eval(&mut scope);
-    simplicity_prog
+    let simplicity_program = simfony_program.eval(&mut scope);
+    let named_commit = simplicity_program
         .finalize_types_main()
-        .expect("Type check error")
+        .expect("Type check error");
```

Comment:
2d68ffbae2d051ebe47756dcfef67448e1ab23df: This error is impossible to reach because translated programs always end in `unit`, as of now.